osgp/protocol-adapter-iec60870/osgp-protocol-simulator-iec60870/src/main/java/org/opensmartgridplatform/simulator/protocol/iec60870/server/Iec60870ConnectionEventListener.java

```diff
@@ -0,0 +1,57 @@
+package org.opensmartgridplatform.simulator.protocol.iec60870.server;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.util.Objects;
+
+import org.openmuc.j60870.ASdu;
+import org.openmuc.j60870.Connection;
+import org.openmuc.j60870.ConnectionEventListener;
+import org.openmuc.j60870.TypeId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Iec60870ConnectionEventListener implements ConnectionEventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Iec60870ConnectionEventListener.class);
+
+    private Iec60870ASduHandlerMap iec60870ASduHandlerMap;
+    private final Connection connection;
+    private final int connectionId;
+
+    public Iec60870ConnectionEventListener(final Connection connection, final int connectionId,
+            final Iec60870ASduHandlerMap iec60870aSduHandlerMap) {
+        this.connection = connection;
+        this.connectionId = connectionId;
+        this.iec60870ASduHandlerMap = iec60870aSduHandlerMap;
+    }
+
+    @Override
+    public void newASdu(final ASdu aSdu) {
+        Objects.requireNonNull(aSdu);
+        try {
+            final TypeId typeId = aSdu.getTypeIdentification();
+            final Iec60870ASduHandler aSduHandler = this.iec60870ASduHandlerMap.getHandler(typeId);
+
+            Objects.requireNonNull(aSduHandler,
+                    String.format("Unknown request received, no handler available for ASdu: %s", aSdu.toString()));
+
+            aSduHandler.accept(this.connection, aSdu);
+
+        } catch (final NullPointerException npe) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)